app/controllers/katello/api/v2/content_views_controller.rb

```diff
@@ -93,11 +93,18 @@ def show
     param :id, :identifier, :desc => N_("content view numeric identifier"), :required => true
     param :name, String, :desc => N_("module name to restrict modules for"), :required => false
     def available_puppet_modules
-      current_uuids = @view.content_view_puppet_modules.where("uuid is NOT NULL").pluck(:uuid)
+      current_cv_puppet_modules = @view.content_view_puppet_modules.where("uuid is NOT NULL")
+      current_uuids = current_cv_puppet_modules.pluck(:uuid)
       repositories = @view.organization.library.puppet_repositories
 
       query = PuppetModule.in_repositories(repositories)
-      query = query.where(:name => params[:name]) if params[:name]
+      if params[:name]
+        query = query.where(:name => params[:name])
+        if current_uuids.present?
+          module_by_name = current_cv_puppet_modules.find_by(:name => params[:name])
+          current_uuids.delete(module_by_name.uuid) if module_by_name && module_by_name.latest_in_modules_by_author?(query)
```

Comment:
@bbuckingham ,

Sorry for repeating the same thing, but could you please confirm, my understanding on below scenarios is correct or not according to your points.

My modifications in code should be in a such a way that,

1) If I select version 'use latest 2.0.1 and click 'select version' then it will show 3 rows ->

a. Use Latest (currently 2.0.1)
b. 2.0.1 
c. 1.3.0
 
2) If I select version '2.0.1' and click 'select version' then it will show 2 rows ->

a. Use Latest (currently 2.0.1)
b. 1.3.0

3) If I select version '1.3.0' and click 'select version' then it will show 2 rows ->

a. Use Latest (currently 2.0.1)
b. 2.0.1